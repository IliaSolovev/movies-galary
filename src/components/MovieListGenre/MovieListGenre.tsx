import React from 'react';

import { Movie } from '../../redux/moviesSlice';

import style from './MovieListGenre.module.scss';

interface Props {
    movie: Movie
}

export const MovieListGenre:React.FC<Props> = ({ movie }) => (
  <div className={style.movieListGenre}>
    {movie === null ? <p>Loading...</p> : (
      <div>
        Films by
        {' '}
        <span className={style.movieListGenre__genre}>
          {movie.genres[0]}
          {' '}
          genre
        </span>
      </div>
    )}

  </div>
);
