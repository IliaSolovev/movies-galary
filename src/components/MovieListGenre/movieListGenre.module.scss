@import "../../styles/variables";
@import "../../styles/main";

.movieListGenre {
  width: 100vw;
  height: 50px;
  background: #e6e6e6;
  font-family: $mainFont;
  font-weight: bold;
  font-size: 1em;
  color: #363636;

  & > div {
    @extend .headerContainer;
    height: 100%;
    display: flex;
    align-items: center;
    @media screen and (max-width: 450px) {
      justify-content: space-around;

    }
  }
  &__genre {
    margin-left: 5px;
    font-weight: bold;
    font-size: .9em;

  }
}
