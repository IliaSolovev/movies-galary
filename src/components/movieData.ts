import { MovieCardData, MovieDescriptionData } from '../types';

export const movieCard2: MovieCardData = {
  id: '337167',
  title: 'Fifty Shades Freed',
  release_date: '2018-02-07',
  poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
  genres: [
    'Drama',
    'Romance',
  ],
};
export const movieCard1: MovieCardData = {
  id: '337167',
  title: 'Fifty Shades Freed',
  release_date: '2019-02-07',
  poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
  genres: [
    'Drama',
    'Romance',
  ],
};

export const movieDescription: MovieDescriptionData = {
  id: '337167',
  title: 'Fifty Shades Freed',
  tagline: "Don't miss the climax",
  rating: 6.1,
  release_date: '2019-02-07',
  poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
  overview: 'Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable '
    + 'connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, '
    + 'new threats could jeopardize their happy ending before it even begins.',
  genres: [
    'Drama',
    'Romance',
  ],
  runtime: 106,
};
