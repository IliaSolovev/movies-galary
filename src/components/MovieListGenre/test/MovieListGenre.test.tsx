import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MovieListGenre } from '../MovieListGenre';
import { movieCard2 } from '../../movieData';


it('should render movie list genre', () => {
  const component = shallow(<MovieListGenre movie={movieCard2} />);
  expect(toJson(component)).toMatchSnapshot();
});
