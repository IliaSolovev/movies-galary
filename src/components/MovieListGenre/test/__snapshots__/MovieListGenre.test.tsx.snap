// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render movie list genre 1`] = `
<div
  className="container"
>
  <div>
    Films by 
    <span
      className="genre"
    >
      Drama
       genre
    </span>
  </div>
</div>
`;
